src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs

```diff
@@ -46,6 +46,39 @@ public virtual bool TryGetFieldType(string name, out FormulaType type)
             return true;
         }
 
+        /// <summary>
+        /// Takes display or logical name as input and returns field and logical name if exists.
+        /// </summary>
+        /// <param name="displayOrLogicalName">Display or Logical name.</param>
+        /// <param name="logical">Logical name for the input.</param>
+        /// <param name="type">Type for the input Display or Logical name.</param>
+        /// <returns>true or false.</returns>
+        /// <exception cref="ArgumentNullException">Throws, if input displayOrLogicalName is empty.</exception>
+        public virtual bool TryGetFieldType(string displayOrLogicalName, out string logical, out FormulaType type)
+        {
+            if (string.IsNullOrEmpty(displayOrLogicalName))
+            {
+                throw new ArgumentNullException("Input parameter \"displayOrLogicalName\" cannot be empty or null");
```

Comment:
>ArgumentNullException [](http://example.com/codeflow?start=26&length=21)

nit:
There's a nameof() operator that returns the string name for a C# identifier. 
And ArgException ctor will provide a message. 

So you can just say:
new ArgumentNullException(nameof(displayOrLogicalName))